proposals/0003.md

```diff
@@ -143,6 +147,32 @@ up by including it a `git commit` alias, for example:
 alias gcs="git commit -s"
 ```
 
+The other way is to configure git to automatically sign-off every commit.
```

Comment:
Could it be perceived negatively if some tool signed off commits for someone outside the core group automatically? I feel like signing something should be a very conscious action for person.

Reply:
I don't understand the issue, could you elaborate? The proposed script sings off with the credentials of the committing person. It's probably not harmful to sign off everything.

Reply:
Sure thing! So the script is signing off with the person's credentials, but they may not be aware of the script doing this for them. So they're not consciously signing the commit themselves -- the script made the decision for them. As a person, for me at least, this comes across as a faux pas. I wouldn't appreciate a script signing things off for me unless I configured it to do so myself, e.g. creating an alias.

Does that make sense?

Maybe this could be alleviated by having a disclaimer in the script and the `CONTRIBUTING.md` file if the repository decides to include the hook?

Reply:
I'm not convinced if that's an issue. We would prefer to have every single commit signed off and the hook reduces the probability of making a mistake. It's not added automatically, so the developer is aware of what's happening. This is an optional step, which must be made consciously.

I personally don't find aliases useful, I'm already using some from oh-my-zsh and that would force me to manually alter them. Also some 3rd party tools can benefit from sing off being a hook and not requiring calling different git commands from usual.

What hit me just now is that the script shouldn't create a global hook, that's way too intrusive. It should just add a few lines to `./.git/hooks/hook-name`.

Reply:
> It's not added automatically, so the developer is aware of what's happening.

Ah, I think I _misunderstood_. I thought that this was advocating to add the hook to our repositories. But it's not, correct? It's just advocating for another way for a contributor to add a sign-off by them adding a hook?

Reply:
Correct! It's a tip for the developers how to configure their git on their machines to always do the right thing

Reply:
That probably means that the text around it isn't quite clear and I need to make it more explicit

Reply:
Sorry for the misunderstanding but glad we cleared it up 😄